pkg/controller.v1beta1/trial/trial_controller_util.go

```diff
@@ -30,14 +30,92 @@ import (
+	if jobStatus.Condition == trialutil.JobSucceeded {
+		if isTrialObservationAvailable(instance) {
+			msg := "Trial has succeeded "
+			reason := TrialSucceededReason
+
+			// Get message and reason from deployed job
+			if jobStatus.Message != "" {
+				msg = fmt.Sprintf("%v. Job message: %v", msg, jobStatus.Message)
+			}
+			if jobStatus.Reason != "" {
+				reason = fmt.Sprintf("%v. Job reason: %v", reason, jobStatus.Reason)
+			}
+
+			instance.MarkTrialStatusSucceeded(corev1.ConditionTrue, reason, msg)
+			instance.Status.CompletionTime = &timeNow
+
+			eventMsg := fmt.Sprintf("Job %v has succeeded", deployedJobName)
+			r.recorder.Eventf(instance, corev1.EventTypeNormal, JobSucceededReason, eventMsg)
+			r.collector.IncreaseTrialsSucceededCount(instance.Namespace)
+		} else {
+			// TODO (andreyvelich): Is is correct to mark succeeded status false when metrics are unavailable?
+			msg := "Metrics are not available"
+			reason := TrialMetricsUnavailableReason
+
+			// Get message and reason from deployed job
+			if jobStatus.Message != "" {
+				msg = fmt.Sprintf("%v. Job message: %v", msg, jobStatus.Message)
+			}
+			if jobStatus.Reason != "" {
+				reason = fmt.Sprintf("%v. Job reason: %v", reason, jobStatus.Reason)
+			}
+
+			instance.MarkTrialStatusSucceeded(corev1.ConditionFalse, reason, msg)
+
+			eventMsg := fmt.Sprintf("Metrics are not available for Job %v", deployedJobName)
+			r.recorder.Eventf(instance, corev1.EventTypeWarning, JobMetricsUnavailableReason, eventMsg)
+		}
+	} else if jobStatus.Condition == trialutil.JobFailed {
+		msg := "Trial has failed"
+		reason := TrialFailedReason
+
+		// Get message and reason from deployed job
+		if jobStatus.Message != "" {
+			msg = fmt.Sprintf("%v. Job message: %v", msg, jobStatus.Message)
+		}
+		if jobStatus.Reason != "" {
+			reason = fmt.Sprintf("%v. Job reason: %v", reason, jobStatus.Reason)
+		}
+
+		instance.MarkTrialStatusFailed(reason, msg)
+		instance.Status.CompletionTime = &timeNow
+
+		eventMsg := fmt.Sprintf("Job %v has failed", deployedJobName)
+		if jobStatus.Message != "" || jobStatus.Reason != "" {
+			eventMsg = fmt.Sprintf("%v. %v %v", eventMsg, jobStatus.Message, jobStatus.Reason)
+		}
+
+		r.recorder.Eventf(instance, corev1.EventTypeNormal, JobFailedReason, eventMsg)
+		r.collector.IncreaseTrialsFailedCount(instance.Namespace)
+	} else if jobStatus.Condition == trialutil.JobRunning {
+		msg := "Trial is running"
+		instance.MarkTrialStatusRunning(TrialRunningReason, msg)
+
+		eventMsg := fmt.Sprintf("Job %v is running", deployedJobName)
+		r.recorder.Eventf(instance, corev1.EventTypeNormal, JobRunningReason, eventMsg)
+		// TODO(gaocegege): Should we maintain a TrialsRunningCount?
+	}
+	// else nothing to do
+	return
+}
+
+// TODO (andreyvelich): Can be deleted after custom CRD is implemented
+func (r *ReconcileTrial) UpdateTrialStatusConditionDeprecated(instance *trialsv1beta1.Trial, deployedJob *unstructured.Unstructured, jobCondition *commonv1.JobCondition) {
```

Comment:
Can we delete it now?

Reply:
@gaocegege We can delete it after we finish Custom CRD implementation.
Controller still runs this function if successCondition and failureCondition is not set here: https://github.com/kubeflow/katib/blob/409b3a2676fa3971acd0850b3bc89655a0ca089e/pkg/controller.v1beta1/trial/trial_controller.go#L307